polaris-react/src/components/IndexTable/components/Cell/Cell.tsx

```diff
@@ -8,18 +8,24 @@ export interface CellProps {
   children?: ReactNode;
   className?: string;
   flush?: boolean;
+  colSpan?: HTMLTableCellElement['colSpan'];
```

Comment:
Looking through the [WCAG spec](https://www.w3.org/WAI/tutorials/tables/multi-level/#table-with-three-headers-related-to-each-data-cell), I think we'll need to add a `header` prop to `Cell` too so that it can render a `th` instead of `td` conditionally 👀 

```suggestion
  colSpan?: HTMLTableCellElement['colSpan'];
  header?: boolean;
}

```